src/Symfony/Component/Validator/ConstraintValidator.php

```diff
@@ -88,6 +88,14 @@ protected function formatValue(mixed $value, int $format = 0): string
             return $value->format('Y-m-d H:i:s');
         }
 
+        if ($value instanceof \UnitEnum) {
+            if ($format & self::OBJECT_TO_STRING) {
+                return $value::class.'::'.$value->name;
```

Comment:
On a closer look, my advice was wrong I think, since the message already contains the class of the enum.
Let's make this be `return 'case '.$value->name;` so that messages looks like this?
"This value should not be identical to Symfony\Component\Validator\Tests\TestEnum case FirstCase"

Reply:
but how can the message know about this? or maybe we should change the type to make it report "enum" and keep this as is?

Reply:
Or maybe formatTypeOf should be changed instead to return `$value::class.'::'.$value->name`?

Reply:
> Or maybe formatTypeOf should be changed instead to return `$value::class.'::'.$value->name`?

I see your point, however, I think that the function should only return the type (the class is the type and is enough for debugging) the same way the formatValue should only return the value (and the name is the textual representation of the enum value). Making each method return one thing whatever is called for. This is why I initially only return the name and probably we could just return to my initial implementation of returning the name of the enum.

Reply:
OK, let's do this.